src/panoptes/utils/horizon.py

```diff
@@ -40,51 +106,45 @@ def __init__(self, obstructions=list(), default_horizon=30):
         """
         super().__init__()
 
-        obstruction_list = list()
-        for obstruction in obstructions:
-            assert isinstance(obstruction, list), "Obstructions must be lists"
-            assert len(obstruction) >= 2, "Obstructions must have at least 2 points"
-
-            obstruction_line = list()
-            for point in obstruction:
-                assert isinstance(point, list), "Obstruction points must be lists"
-                assert len(point) == 2, "Obstruction points must be 2 points"
+        # Parse obstruction list
+        if obstructions is None:
+            obstructions = []
+        self.obstructions = [Obstruction(_) for _ in obstructions]
 
-                assert type(point[0]) is not bool, "Bool not allowed"
-                assert type(point[1]) is not bool, "Bool not allowed"
-                assert isinstance(point[0], (float, int, np.integer)), "Must be number-like"
-                assert isinstance(point[1], (float, int, np.integer)), "Must be number-like"
+        # Add default horizon
+        self._default_horizon = get_quantity_value(default_horizon, "deg") * u.deg
 
-                alt = float(point[0])
-                az = float(point[1])
+        # Calculate horizon at each integer azimuth
+        # This is included for backwards compatibility with POCS
+        self.horizon_line = np.zeros(360, dtype="float")
+        for i, az in enumerate(self.horizon_line):
+            self.horizon_line[i] = self.get_horizon(az).to_value(u.deg)
 
-                assert 0. <= alt <= 90., "Altitude must be between 0-90 degrees"
-
-                if az < 0.:
-                    az += 360
+    def get_horizon(self, az):
+        """ Get the horizon level in degrees at a given azimuth.
+        Args:
+            az (float or astropy.Quantity): The azimuth. If float, assumed in degrees.
+        Returns:
+            astropy.Quantity: The angular horizon level.
+        """
+        az = get_quantity_value(az, "deg") * u.deg
 
-                assert 0. <= az <= 360., "Azimuth must be between 0-360 degrees"
-                obstruction_line.append((alt, az))
-            obstruction_list.append(sorted(obstruction_line, key=lambda point: point[1]))
+        horizon = self._default_horizon
 
-        self.obstructions = sorted(obstruction_list, key=lambda point: point[1])
+        for obstruction in self.obstructions:
+            horizon = max(horizon, obstruction.get_horizon(az))
 
-        self.default_horizon = default_horizon
-        self.horizon_line = np.ones(360) * self.default_horizon
+        return horizon
 
-        # Make helper lists of the alt and az
-        self.alt = list()
-        self.az = list()
-        for obstruction in self.obstructions:
-            self.alt.append([point[0] for point in obstruction])
-            self.az.append([point[1] for point in obstruction])
+    def make_plot(self):
```

Comment:
As mentioned above, I'd rather not have plotting be a part of the main class (unless we are going all the way and integrate it into the doctests).

Reply:
I've tested the notebook I provided and it still works (and is actually fixed)!